sigridci/sigridci.py

```diff
@@ -606,9 +620,24 @@ def displayMetadata(self, apiClient, options):
             for key, value in apiClient.fetchMetadata().items():
                 if value:
                     print(f"    {key}:".ljust(20) + str(value))
+                    
+    def prepareMetadata(self, options):
+        metadata = {field: os.environ.get(field.lower(), "") for field in self.METADATA_FIELDS}
+        metadata = {field: value for field, value in metadata.items() if value and len(value) > 0}
```

Comment:
Is this common Python idiom? We first initialize `metadata` in line 625 and then apply some filter-like construct in this line. Would it be possible to do this all at once, a bit like how you'd do this with Java's streams?

Reply:
You can, but then the line would be very long. The downside is that you would need to repeat the logic both on the left of the dict comprehension and on the filter:

    metadata = {field: os.environ.get(field.lower(), "") for field in self.METADATA_FIELDS if os.environ.get(field.lower(), "")}

Alternatively, this would probably be nicer:

    getMetadataValue = lambda field: os.environ.get(field.lower(), "")
    metadata = {field: getMetadataValue(field) for field in self.METADATA_FIELDS if getMetadataValue(field)}